src/designspace.rs

```diff
@@ -0,0 +1,213 @@
+    pub filename: String,
+    /// The name of the layer in the source file. If no layer attribute is given assume the foreground layer should be used.
+    pub layer: Option<String>,
+    /// Location in designspace coordinates.
+    pub location: Location,
+}
+
+/// https://fonttools.readthedocs.io/en/latest/designspaceLib/xml.html#instances-element
+#[derive(Clone, Debug, Default, PartialEq, Deserialize)]
+#[serde(rename = "instances")]
+pub struct Instances {
+    /// One or more instances located somewhere in designspace.
+    pub instance: Vec<Instance>,
+}
+
+/// An [instance]].
+///
+/// [instance]: https://fonttools.readthedocs.io/en/latest/designspaceLib/xml.html#instance-element
+#[derive(Clone, Debug, Default, PartialEq, Deserialize)]
+pub struct Instance {
+    // per @anthrotype, contrary to spec, filename, familyname and stylename are optional
+    /// The family name of the instance font. Corresponds with font.info.familyName
+    pub familyname: Option<String>,
+    /// The style name of the instance font. Corresponds with font.info.styleName
+    pub stylename: Option<String>,
+    /// A unique name that can be used to identify this font if it needs to be referenced elsewhere.
+    pub name: String,
+    /// A path to the instance file, relative to the root path of this document. The path can be at the same level as the document or lower.
+    pub filename: Option<String>,
+    /// Corresponds with font.info.postscriptFontName
+    pub postscriptfontname: Option<String>,
+    /// Corresponds with styleMapFamilyName
+    pub stylemapfamilyname: Option<String>,
+    /// Corresponds with styleMapStyleName
+    pub stylemapstylename: Option<String>,
+    /// Location in designspace.
+    pub location: Location,
+}
+
+/// A [design space location]].
+///
+/// [design space location]: https://fonttools.readthedocs.io/en/latest/designspaceLib/xml.html#location-element-source
+#[derive(Clone, Debug, Default, PartialEq, Deserialize)]
+pub struct Location {
+    /// Should contain one Dimension for each axis in the designspace.
+    pub dimension: Vec<Dimension>,
+}
+
+/// A [design space dimension]].
+///
+/// [design space location]: https://fonttools.readthedocs.io/en/latest/designspaceLib/xml.html#location-element-source
+#[derive(Clone, Debug, Default, PartialEq, Deserialize)]
+pub struct Dimension {
+    /// Name of the axis, e.g. Weight.
+    pub name: String,
+    /// Value on the axis in user coordinates.
+    pub uservalue: Option<f32>,
+    /// Value on the axis in designcoordinates.
+    pub xvalue: Option<f32>,
+    /// Separate value for anisotropic interpolations.
+    pub yvalue: Option<f32>,
+}
+
+impl DesignSpaceDocument {
+    /// Load a designspace.
+    pub fn load<P: AsRef<Path>>(path: P) -> Result<DesignSpaceDocument, DesignSpaceLoadError> {
+        let reader = BufReader::new(File::open(path).map_err(DesignSpaceLoadError::Io)?);
+        from_reader(reader).map_err(DesignSpaceLoadError::DeError)
```

Comment:
(an observation, not a suggestion:) there's a common convention in rust projects to rarely import free functions directly, but to call them via the containing module/crate; this has the advantage of making it clear when a function is external or part of the current module.

Reply:
Oh that's a nice convention. Updated to use qualified name.